pkg/graveler/ref/commit_iterator.go

```diff
@@ -115,42 +118,75 @@ func (ci *CommitIterator) Next() bool {
 			ci.queue.Push(rec)
 		}
 	}
-
-	if ci.queue.Len() == 0 {
-		ci.value = nil
-		ci.state = commitIteratorStateDone
-		return false
-	}
-
-	// as long as we have something in the queue we will
-	// set it as the current value and push the current commits parents to the queue
-	ci.value = heap.Pop(&ci.queue).(*graveler.CommitRecord)
-	parents := ci.value.Parents
-	if ci.firstParent && len(parents) > 1 {
-		parents = parents[:1]
-	}
-	for _, p := range parents {
-		// skip commits we already visited
-		if _, visited := ci.visit[p]; visited {
+	// running as long as we have commit in queue, popping one each iteration
+	for ci.queue.Len() > 0 {
+		ci.value = heap.Pop(&ci.queue).(*graveler.CommitRecord)
+
+		// case --no-merges flag, and there is more than one parent, treating only the first one
+		if ci.noMerges && len(ci.value.Parents) > 1 {
+			firstParent := ci.value.Parents[0]
+
+			// case parent was visited, continuing.
```

Comment:
I don't understand.
1. Here no-merges implies first-parent. Why? I would expect no-merges to do just one thing. 
1. How does this code work when there are 2 merges one after the other? firstParent Will also be a merge and the code should skip it.
1. (Related) Just skip the current record of its a merge and keep going. This is a filter operation, let the next iteration do its thing